### PR TITLE
[web][camera] Fixed issue with facing on web mobile (30222)

### LINKED ISSUE #30222: Expo-camera does not allow to use 'back' camera when using iPhone/Android web version from Chrome/Safari
### Minimal reproducible example

https://github.com/NazarKozak/Expo-camera-bug

### What platform(s) does this occur on?

Web

### Did you reproduce this issue in a development build?

No (web-only issue)

### Summary

Is expected to render back camera view when defaulting to 'back' CameraType or flipping from 'front' to 'back'

### Environment

expo-camera: 15.0.3
react-native: 0.74.1
react-native-web: 0.19.6

### Expo Doctor Diagnostics

✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!

### PR DESCRIPTION
**Pull request by entiendoNull**
Fixes #30222

# Why

This PR fixes the issue announced here; https://github.com/expo/expo/issues/30222
"Expo-camera does not allow to use 'back' camera when using iPhone/Android web version from Chrome/Safari"

# How

I have followed the steps described here in order to contribute.
https://github.com/expo/expo/blob/main/CONTRIBUTING.md

I spun the project up locally, and was able to reproduce the issue that was filed. I tracked the problem down to a prop (`type`) that was no longer applicable (it's now `facing`).

The web integration of the latest version of the camera seem to refer to some legacy types. 
(And on a sidenote;  the legacy version seem to refer to some files outside of it's legacy scope/directory.)

I changed the `CameraNativeProps` to not refer to the legacy version anymore. This enabled me to use `facing` instead of `type`.

Changing the type reference to use the current, rather than the legacy also revealed a few other issues, such as some naming changes to certain props e.g;
barCodeScannerSettings -> barcodeScannerSettings
onBarCodeScanned -> onBarcodeScanned

The `interval` property in the `BarcodeSettings` seems to have been removed in the new version. However, I could not get rid of the interval all together as both the new camera and the legacy uses the same version of `useWebQRScanner`. For the new camera I set the interval to `300` - this may be something that you need to evaluate. 

I have run yarn build, lint --fix, test

I suppose the web sdk for this module may need some attention, but this PR should resolve this issue in the meantime.

# Test Plan

- I have tested it on Safari on my phone, and it's now possible to change the facing (before it wasn't).
- I have tested it on ExpoGo (iOS), and my changes seem to not have affected it (which was expected).
- I have tested on my laptop (Chrome/Safari/Firefox) and the facing is always "front" (which was expected)

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-camera/src/ExpoCamera.web.tsx
Comment (lukmccall) [inline packages/expo-camera/src/ExpoCamera.web.tsx:52]: I think we should use the prop here. Hardcoding it to 300 doesn't look right.

**Review by lukmccall — APPROVED** [review]
Looking good! Thanks for your contribution. I've left one comment.

File: packages/expo-camera/src/ExpoCamera.web.tsx
Comment (entiendoNull) [reply packages/expo-camera/src/ExpoCamera.web.tsx:52]: Hi @lukmccall! 
Thank you for your comment! :D Glad I can be able to contribute.
I do agree, but decided to remove it as;

- the interval prop is not covered in the docs
- the interval prop is not covered in the new type declarations

The `interval` prop does exist on the legacy camera.
Thus, I wrote this text in my PR.

```
The interval property in the BarcodeSettings seems to have been removed in the new version. However, I could not get rid of the interval all together as both the new camera and the legacy uses the same version of useWebQRScanner. For the new camera I set the interval to 300 - this may be something that you need to evaluate.
```

I'm not sure why it was decided it should be removed, but I decided to accept the fact that it's no longer supported and thus added a hard coded value (which may need to be revised).

Comment (entiendoNull) [reply packages/expo-camera/src/ExpoCamera.web.tsx:52]: I believe the legacy version of expo-camera should be removed in SDK 52 - if I don't misremember the info from the blog posts. With that change, I believe some (of at least the web stuff) of this package will receive some further attention. 

This PR will currently solve a few issues apart from the reported main issue (e.g. casing of prop names etc). If the `interval`-prop should be reintroduced, perhaps it's better to open a specific PR for it where docs (- and every other affected part) are touched accordingly)?

Comment (entiendoNull) [reply packages/expo-camera/src/ExpoCamera.web.tsx:52]: @lukmccall, I did not change the `interval` (see the other posts above) but saw you approved of the changes. I rebuilt the project as this was feedback I received from the workflows. 
To re-run the workflows it need a maintainer approval. Is that something you can help me with?

**Comment by henrilp** [comment]
CameraView is not detecting qr codes on firefox.

**Pull request closed without merging**
